app/controllers/user_videos_controller.rb

```diff
@@ -3,10 +3,12 @@ def new; end
 
   def create
     user_video = UserVideo.new(user_video_params)
-    if current_user.user_videos.find_by(video_id: user_video.video_id)
+    if current_user && current_user.user_videos.find_by(video_id: user_video.video_id)
       flash[:error] = 'Already in your bookmarks'
     elsif user_video.save
       flash[:success] = 'Bookmark added to your dashboard!'
+    else
+      flash[:notice] = 'User must login to bookmark videos.'
```

Comment:
Flash message telling visitor to login rather than redirecting them.
I am hoping to find a JS solution for this.

Reply:
Cool looks good. JS would be interesting